### ISSUE TITLE
Minimax strategy and early_stopping

### ISSUE THREAD

**Issue by ohshushu** [thuml/Anomaly-Transformer#45]
Hi, I have two issues with Minimax strategy and early stopping :
1, loss1 is to maximize series_association and loss2 is to minimize prior_association. In the original paper, it was minimize, then maximize. Why did it become maximize, then minimize in the code?
2. Why score = -loss when stopping early？

**Comment by ruinski** [comment]
hi, i have same issue.
if the window size = 100 in single sequence data, the early stopping work in epoch = 1.
so the model doesn't learning dataset. the loss function wasn't work

**Comment by zixuan-go** [comment]
Come uninvited : ). 
The following is my own understanding, for reference only, correctness not guaranteed：
*Please note that the **order in the code is Max-Min**, not **Min-Max as in the paper.** 

The reason why Min-Max cannot be used is that **the random initialization of series association (SA) has no meaning** and may be very far away from the target point, which makes it difficult to continue training. **In extreme cases, the model can no longer use more distant context for modeling.** 

The **initialization of prior association (PA) is affected by the unimodal statistical properties of its Gaussian kernel function, so there is a natural neighborhood meaning** and it can be directly used as the **Max-training label**.

That's why we use Max-Min rather Min-Max. Hope for helping.
